task_21/Aleksandr.Timoshenko/src/test/java/com/hillel/tasks/TrainTeast.java

```diff
@@ -0,0 +1,65 @@
+package com.hillel.tasks;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class TrainTeast {
+
+  Train train;
+
+  @Before
+  public void setUp() {
+    train = new Train(3);
+  }
+
+  @Test(expected = IndexOutOfBoundsException.class)
+  public void isTicketInStockTest() {
+    assertThat(train.isTicketInStock(4, 4), is(4));
+  }
+
+  @Test(expected = IndexOutOfBoundsException.class)
+  public void isTicketInStockTestToo() {
+    assertThat(train.isTicketInStock(2, 44), is(4));
+  }
+
+  @Test
+  public void isTicketInStockTestThree() {
+    boolean result = train.isTicketInStock(2, 4);
+    assertThat(result, is(true));
+  }
+
+  @Test
+  public void isTicketInStockTestFour() throws NoTicketException {
+    train.soldTicket(2, 4);
+    boolean result = train.isTicketInStock(2, 4);
+    assertThat(result, is(false));
+  }
+
+  @Test(expected = NoTicketException.class)
+  public void isTicketInStockTestFive() throws NoTicketException {
+    try {
+      train.soldTicket(2, 4);
+      train.soldTicket(2, 4);
+    } catch (NoTicketException e) {
+      throw new NoTicketException(e.getMessage());
```

Comment:
Rethrowing exception here makes no sense. It is better to check in test a condition where it will be thrown and another test where there will be no exception